pkg/datapath/loader/cache.go

```diff
@@ -222,12 +222,17 @@ func (o *objectCache) build(ctx context.Context, cfg *templateCfg, hash string)
 //
 // Returns the path to the compiled template datapath object and whether the
 // object was compiled, or an error.
-func (o *objectCache) fetchOrCompile(ctx context.Context, cfg datapath.EndpointConfiguration, stats *SpanStat) (string, bool, error) {
-	hash, err := o.baseHash.sumEndpoint(o, cfg, false)
+func (o *objectCache) fetchOrCompile(ctx context.Context, cfg datapath.EndpointConfiguration, stats *SpanStat) (path string, compiled bool, err error) {
+	var hash string
+	hash, err = o.baseHash.sumEndpoint(o, cfg, false)
 	if err != nil {
 		return "", false, err
 	}
 
+	// Capture the time spent waiting for the template to compile.
+	stats.bpfWaitForELF.Start()
+	defer stats.bpfWaitForELF.End(err == nil)
```

Comment:
Does this work for the `!ok` branch below?

Reply:
ugh, apparently not. I thought that because the named return value is `err` then the `err` would be populated before the defer is run, but that does not appear to be the case.

That said, we completely disable the labels when endpoint regeneration fails here:

https://github.com/cilium/cilium/blob/bc09f74d22cfb1ee3604b849a631a6fbcb2ea221/pkg/endpoint/metrics.go#L57-61

I'll fix it up, good spotting.

Reply:
For anyone following along at home, the error here is that the parameters to `stats.bpfWaitForELF.End(...)` are calculated at this point, then the usage of those parameters is deferred.